src/openzaak/components/documenten/api/filters.py

```diff
@@ -48,26 +45,13 @@ class Meta:
         }
 
 
-def object_queryset(request):
-    object_value = request.query_params.get("object", "")
-    object_path = urlparse(object_value).path
-
-    # get main_object data formatted by serializer
-    try:
-        viewset = get_viewset_for_path(object_path)
-    except ObjectDoesNotExist:
-        # the exception is raised in URLModelChoiceField.to_python method
-        return ObjectInformatieObject.objects.all()
-    model = viewset.get_queryset().model
-    return model._default_manager.all()
-
-
-class ObjectFilter(URLModelChoiceFilter):
+class ObjectFilter(FkOrUrlFieldFilter):
     def filter(self, qs, value):
-        if isinstance(value, Zaak):
-            self.field_name = "zaak"
-        elif isinstance(value, Besluit):
+        path = urlparse(value).path
+        if "/besluiten/" in path:
```

Comment:
Don't like these hard coded URLs. Can't you use `vng_api_common.utils.get_viewset_for_path` instead for this? You would then be able to figure out the model from `viewset.queryset.model` and map that to the relevant field.

Reply:
But if the path to the external url would differ from openzaak url patterns it couldn't be resolved
For example, the path: "https://externe.zaken.nl/api/v1/zaken/{uuid}" can't be resolved with `get_viewset_for_path` because the url lacks subpath `zaken`.

Reply:
Right, I missed that. Can you add that as a comment? I'd also do stricter validation then - `{whatever}/besluiten/{uuid}` should match as pattern

Reply:
a comment and stricter validation are added